ortb/clone.go

```diff
@@ -0,0 +1,271 @@
+package ortb
+
+import (
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/util/ptrutil"
+	"github.com/prebid/prebid-server/util/sliceutil"
+)
+
+func CloneApp(s *openrtb2.App) *openrtb2.App {
+	if s == nil {
+		return nil
+	}
+
+	// Shallow Copy (Value Fields)
+	c := *s
+
+	// Deep Copy (Pointers)
+	c.Cat = sliceutil.Clone(s.Cat)
+	c.SectionCat = sliceutil.Clone(s.SectionCat)
+	c.PageCat = sliceutil.Clone(s.PageCat)
+	c.Publisher = ClonePublisher(s.Publisher)
+	c.Content = CloneContent(s.Content)
+	c.KwArray = sliceutil.Clone(s.KwArray)
+	c.Ext = sliceutil.Clone(s.Ext)
```

Comment:
Isn't `Ext` just a `json.RawMessage`? I suppose slice clone works because it is a slice of bytes. But we shouldn't have to copy this in the same way we aren't copying string fields. And the slice clone is probably not the most efficient way to copy a `json.RawMessage` if we do decide we need a full copy of this. This goes for all the following `Ext` fields as well.

Reply:
> I suppose slice clone works because it is a slice of bytes.

Exactly. The slice clone works fine because the type is really just `[]byte`.  We don't need to clone string fields because strings are immutable. The `ext` clones are needed because I plan to perform an in-place `json.Unmarshal` which can reuse the backing array of the slice.

> And the slice clone is probably not the most efficient way to copy a json.RawMessage

Do you know of a better way?

Reply:
Here's [an example](https://go.dev/play/p/UhIn4WuO4Zu) demonstrating the original slice is affected by unmarshal.